File: Neetcode/MinRotateSortArray.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left = 0
        right = len(nums) - 1
        minelem = 5001
        if nums[left] < nums[right]:
            return nums[left]
        while left <= right:
            mid = left + (right - left)//2
            minelem = min(minelem, nums[mid])
            if nums[mid] < nums[right]:
                right = mid - 1
            else:
                left = mid + 1
        return minelem

File: Neetcode/MinRotateSortArray.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int left = 0;
        int right = nums.size() - 1;
        int mid = 0;
        while (left <= right){
            mid = left + (right - left)/2;
            if (left == right){
                break;
            }
            cout << mid;
            if (nums[mid] > nums[right]){
                left = mid + 1;
            }
            else if (nums[mid] < nums[right]){
                right = mid;
            }
        }
        return nums[mid];
    }
};
